### ISSUE TITLE
@ character in username breaks the connection string

### ISSUE THREAD

**Issue by paoliniluis** [metabase/metabase#22862]
**Describe the bug**
"@" character in the database username breaks the conn string in MB_DB_CONNECTION_URI. Usernames in the form of "user@host" is the standard in Azure Postgres

**Logs**
Any conn string that had a @ before 43 will now error out when Metabase wants to connect to it depending on where the @ is. e.g.
WARN postgresql.Driver :: JDBC URL must contain a / at the end of the host or port: jdbc:postgresql://&password=mysecretpassword&ssl=true&sslmode=require

**To Reproduce**
Use the following docker-compose file
<details>
```
version: '3.9'
services:
  metabase:
    image: metabase/metabase:v0.43.1
    container_name: metabase
    hostname: metabase
    volumes: 
    - /dev/urandom:/dev/random:ro
    ports:
      - 3001:3000
    environment: 
      - "MB_DB_CONNECTION_URI=postgres://postgres-app-db:5432/metabase?user=metabase@postgres-app-db&password=mysecretpassword"
    cpus: 1
    mem_limit: 2048m
    networks: 
      - metanet1
    depends_on: 
      - postgres-app-db
  postgres-app-db:
    image: postgres:14.2-alpine
    container_name: postgres-app-db
    hostname: postgres-app-db
    environment:
      - "POSTGRES_USER=metabase@postgres-app-db"
      - "POSTGRES_DB=metabase"
      - "POSTGRES_PASSWORD=mysecretpassword"
    networks: 
      - metanet1
    cpus: 1
    mem_limit: 128m
networks: 
  metanet1:
    driver: bridge
```
</details>

save the file and do a "docker-compose up" on the directory you saved the file, see the error
2022-05-19 19:42:33,680 WARN postgresql.Driver :: JDBC URL must contain a / at the end of the host or port: jdbc:postgresql://postgres-app-db&password=mysecretpassword

If you change the MB_DB_CONNECTION_URI to "postgres://metabase@postgres-app-db:mysecretpassword@postgres-app-db:5432/metabase" it won't be able to find the host

**Expected behavior**
Either fix this little issue with @ in usernames so Metabase parses them correctly or we have to include this in the documentation

**Screenshots**
NA

**Information about your Metabase Installation:**
- Your operating system: Pop-os 22.04
- Metabase version: 0.43.1
- Metabase hosting environment: Docker

**Severity**
None, has a workaround. You need to use the conn string as "MB_DB_CONNECTION_URI=postgres://host:port/dbname?user=user@host&password=password" and replace the user@host for user%40host so in the previous docker-compose file it would be "postgres://postgres-app-db:5432/metabase?user=metabase%40postgres-app-db&password=mysecretpassword"

**Additional context**
Reported in https://github.com/metabase/metabase/issues/22814, credits to @dhmoto14

**Comment by camsaul** [comment]
In #20122 we added support for passing `MB_DB_USER` and/or `MB_DB_PASS` separately from `MB_DB_CONNECTION_URI`. If you can't pass us a valid JDBC connection URI then you can always pass in user and password separately. We're not going back to parsing the connection string.

Closing as Won't Fix since #20122 is in place as a workaround for edge cases like this

**Comment by flamber** [comment]
@adamfeldman The release notes were too big to include all the 250+ issues addressed. Here's the full list: https://github.com/metabase/metabase/milestone/149?closed=1
